packages/node/src/indexer/api.service.ts

```diff
@@ -86,20 +99,45 @@ export class ApiService
 
     this.registry = await this.buildRegistry();
 
-    await this.createConnections(
-      network,
-      (endpoint) =>
-        CosmosClientConnection.create(
-          endpoint,
-          this.fetchBlocksBatches,
-          this.registry,
-        ),
-      (connection: CosmosClientConnection) => {
-        const api = connection.unsafeApi;
-        return api.getChainId();
-      },
-    );
+    if (this.nodeConfig.kyveEndpoint) {
+      // still need to use cosmosClient to proxy rpcCalls
+      const cosmosClient = await CosmosClientConnection.create(
+        (network.endpoint as string[])[0],
+        this.fetchBlocksBatches,
+        this.registry,
+      );
+
+      await this.createConnections(
+        network,
+        (endpoint) =>
+          KyveConnection.create(
+            this.nodeConfig.kyveEndpoint,
+            network.chainId,
+            this.registry,
+            this.nodeConfig.storageUrl,
+            this.nodeConfig.kyveChainId,
+            cosmosClient,
+            this.project.root,
+          ),
+        (connection: KyveConnection) => Promise.resolve(network.chainId),
+      );
+    } else {
+      await this.createConnections(
+        network,
+        (endpoint) =>
+          CosmosClientConnection.create(
+            endpoint,
+            this.fetchBlocksBatches,
+            this.registry,
+          ),
+        (connection: CosmosClientConnection) => {
+          const api = connection.unsafeApi;
+          return api.getChainId();
+        },
+      );
+    }
```

Comment:
This isn't the right way to do it. It would result in every url having a new Kyve connection

```ts
await this.createConnections(
      network,
      (endpoint) => {
        if (endpoint == kyveEndpoint) return KyveConnection.create(....)
        return CosmosClientConnection.create(
          endpoint,
          this.fetchBlocksBatches,
          this.registry,
        ),
     }
      (connection: CosmosClientConnection) => {
        const api = connection.unsafeApi;
        return api.getChainId();
      },
    );
```